march_hardware/src/IMotionCube.cpp

```diff
@@ -106,6 +113,31 @@ void IMotionCube::actuateRad(float targetRad)
   this->actuateIU(this->encoder.RadtoIU(targetRad));
 }
 
+void IMotionCube::actuateCurrent(float targetCurrent)
+{
+  ROS_ASSERT_MSG(this->actuationMode == ActuationMode::torque, "trying to actuate current, while actuationmode = "
+                                                               "%s",
+                 this->actuationMode.toString().c_str());
+
+  ROS_ASSERT_MSG(targetCurrent < 20000, "Current of %f is too high.", targetCurrent);
+
+  union bit16 targetCurrentStruct;
+  targetCurrentStruct.i = targetCurrent;
```

Comment:
Storing a float as int

Reply:
Yeah if it really is a float, this won't work. You screw up the variable if you interpret a floating point variable as an int.